resolwe_bio/management/commands/generate_samples.py

```diff
@@ -0,0 +1,248 @@
+import gzip
+import io
+import json
+import os
+import random
+import shutil
+import string
+import zipfile
+
+from django.core.management.base import BaseCommand
+from resolwe.flow.models import Data, DescriptorSchema, Process, Collection, Storage
+from django.contrib.auth.models import User
+from django.conf import settings
+
+
+class Command(BaseCommand):
+
+    """Generate test data"""
+
+    help = 'Generate test data'
+
+    def add_arguments(self, parser):
+        parser.add_argument('-a', '--n-annotated', type=int,
+                            required=False, default=15, help="Number of annotated samples to generate")
+        parser.add_argument('-u', '--n-unannotated', type=int,
+                            required=False, default=5, help="Number of unannotated samples to generate")
+
+    def word(self, length):
+        return ''.join(random.choice(string.lowercase) for i in range(length))
+
+    def set_name(self):
+        organism = random.choice(['Dictyostelium discoideum', 'Mus musculus', 'Homo sapiens'])
+        replicate = random.choice(['rep1', 'rep2', 'rep3', 'rep4', 'rep5'])
+        hour = random.choice(range(36))
+        kit = random.choice(['RiboZero', 'Nugen'])
+        group = random.choice(['treatment', 'control'])
+        if organism == 'Dictyostelium discoideum':
+            return 'Dd_{}_{}_hr_{}'.format(kit, replicate, hour)
+        if organism == 'Mus musculus':
+            return 'Mm_{}_{}_{}'.format(self.word(3), group, replicate)
+        if organism == 'Homo sapiens':
+            return 'Hs_{}_{}_{}'.format(self.word(3), group, replicate)
+
+    def gzopen(self, fname):
+        """Open Gzip files using io.BufferedReader."""
+        return io.TextIOWrapper(io.BufferedReader(gzip.open(fname)))
+
+    def escape_mongokey(self, key):
+        """Escape keys when serializing database entries"""
+        return key.replace('$', u'\uff04').replace('.', u'\uff0e').replace(' ', '_')
+
+    def generate_expressions(self, gene_ids, path):
+        """Generate random expression data"""
+        genes = {}
+        with gzip.open(os.path.join(path, 'expressions.tab.gz'), 'wb') as f:
+            f.write('{}\t{}\n'.format('Gene', 'Expression'))
+            with self.gzopen(gene_ids) as gene_ids:
+                all_genes = [line.strip() for line in gene_ids]
+                for g in all_genes:
+                    e = random.gammavariate(1, 100)
+                    f.write('{}\t{}\n'.format(g, e))
+                    genes[self.escape_mongokey(g)] = e
+
+        with open(os.path.join(path, 'expressions.json'), 'w') as j:
+            js_out = '{"exp_json":%s}' % json.dumps({'genes': genes}, separators=(',', ':'))
```

Comment:
Use `format` instead of `%`, use double brackets if want want bracket in output:

`js_out = '{{"exp_json":{}}}'.format(json.dumps({'genes': genes}, separators=(',', ':')))`